app/HutchManager/Services/ResultsTranslator.cs

```diff
@@ -7,36 +9,109 @@ public class ResultsTranslator
   
   public interface IResultsTranslator<T>
   {
-    QueryResult TranslateRoCrates(T input);
+    T TranslateRoCrates(ROCratesQueryResult input);
   }
   
-  public class RoCratesQueryTranslator : IResultsTranslator<ROCratesQueryResult>
+  public class RoCratesQueryTranslator : IResultsTranslator<QueryResult>
   {
 
-    public  QueryResult TranslateRoCrates(ROCratesQueryResult job)
+    public QueryResult TranslateRoCrates(ROCratesQueryResult job)
     {
       var rquestQueryResult = new QueryResult();
       foreach (var graph in job.Graphs)
       {
-        switch (graph.Name)
+        var property = SchemaSerializer.DeserializeObject<PropertyValue>(graph.ToString());
+        switch (property.Name)
         {
           case "activity_source_id":
-            int activitySourceId = Int32.Parse(graph.Value);
+            int activitySourceId = int.Parse(property.Value);
             rquestQueryResult.ActivitySourceId = activitySourceId;
             break;
           case "job_id":
-            var jobId = graph.Value;
+            var jobId = property.Value;
             rquestQueryResult.JobId = jobId;
             break;
           case "count":
-            int count = Int32.Parse(graph.Value);
+            int count = int.Parse(property.Value);
             rquestQueryResult.Results.Count = count;
             break;
         }
-
       }
       return rquestQueryResult ;
     }
+  }
 
+  public class RoCratesToDistribution : IResultsTranslator<DistributionQueryTaskResult>
+  {
+    public DistributionQueryTaskResult TranslateRoCrates(ROCratesQueryResult job)
+    {
+      var rquestQueryResult = new DistributionQueryTaskResult();
+      foreach (var graph in job.Graphs)
+      {
+        if (graph.TryGetProperty("@type", out var type))
+        {
+          switch (type.ToString())
+          {
```

Comment:
need to fix all the null reference warnings.

You can do this one of the following ways:

- explicitly allowing it to be null, if appropriate
- by convincing the compiler it'll never be null (using null checks)
- accepting the possibility of null and using defaults to account for nulls
- telling the compiler to shut up even if it might be null (using the `!`) if appropriate